File: Java/BOJ/스택_10828.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;
import java.util.Stack;

public class Ω∫≈√_10828 {
	public static void main(String args[]) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int count = Integer.parseInt(br.readLine());

		Stack stk = new Stack();
		
		for(int i=0; i<count; i++) {
			
			StringTokenizer st = new StringTokenizer(br.readLine(), " ");
			String type = st.nextToken();
			
			if(type.equals("push")) {
				stk.push(Integer.parseInt(st.nextToken()));
			} else if(type.equals("pop")) {
				System.out.println(stk.isEmpty() ? "-1" : stk.pop());
			}  else if(type.equals("size")) {
				System.out.println(stk.size());
			} else if(type.equals("empty")) {
				System.out.println(stk.isEmpty() ? "1" : "0");
			} else if(type.equals("top")) {
				System.out.println(stk.isEmpty() ? "-1" : stk.peek());
			}
		}
		br.close();
	}
}

File: Kotlin/BOJ/스택_10828.kt
package com.example.kotlincodingtest.baekjoon

import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter
import java.util.Stack
import java.util.StringTokenizer

fun main() = with(BufferedReader(InputStreamReader(System.`in`))){
    val bw = BufferedWriter(OutputStreamWriter(System.`out`))
    val stack = Stack<Int>()
    val num: Int = readLine().toInt()
    for(n in 0 until num) {
        val st = StringTokenizer(readLine(), " ")
        when(st.nextToken()) {
            "push" -> stack.push(st.nextToken().toInt())
            "pop" -> bw.write(if(stack.empty()) "-1\n" else "${stack.pop()}\n")
            "size" -> bw.write("${stack.size}\n")
            "empty" -> bw.write(if(stack.empty()) "1\n" else "0\n")
            "top" -> bw.write(if(stack.empty()) "-1\n" else "${stack.peek()}\n")
        }
    }
    bw.flush()
    bw.close()
    close()
}